statsmodels/tsa/arima/estimators/hannan_rissanen.py

```diff
@@ -0,0 +1,196 @@
+
+Author: Chad Fulton
+License: BSD-3
+"""
+import numpy as np
+
+from scipy.signal import lfilter
+from statsmodels.tools.tools import Bunch
+from statsmodels.regression.linear_model import OLS, yule_walker
+from statsmodels.tsa.tsatools import lagmat
+
+from statsmodels.tsa.arima.specification import SARIMAXSpecification
+from statsmodels.tsa.arima.params import SARIMAXParams
+
+
+def hannan_rissanen(endog, ar_order=0, ma_order=0, demean=True,
+                    initial_ar_order=None, unbiased=True):
+    """
+    Estimate ARMA parameters using Hannan-Rissanen procedure.
+
+    Parameters
+    ----------
+    endog : array_like
+        Input time series array, assumed to be stationary.
+    ar_order : int
+        Autoregressive order
+    ma_order : int
+        Moving average order
+    demean : bool, optional
+        Whether to estimate and remove the mean from the process prior to
+        fitting the ARMA coefficients. Default is True.
+    initial_ar_order : int, optional
+        Order of long autoregressive process used for initial computation of
+        residuals.
+    unbiased: bool, optional
+        Whether or not to apply the bias correction step. Default is True.
+
+    Returns
+    -------
+    parameters : SARIMAXParams object
+    other_results : Bunch
+        Includes three components: `spec`, containing the
+        `SARIMAXSpecification` instance corresponding to the input arguments;
+        `initial_ar_order`, containing the autoregressive lag order used in the
+        first step; and `resid`, which contains the computed residuals from the
+        last step.
+
+    Notes
+    -----
+    The primary reference is [1]_, section 5.1.4, which describes a three-step
+    procedure that we implement here.
+
+    1. Fit a large-order AR model via Yule-Walker to estimate residuals
+    2. Compute AR and MA estimates via least squares
+    3. (Unless `unbiased=False`) Perform bias correction
+
+    The order used for the AR model in the first step may be given as an
+    argument. If it is not, we compute it as suggested by [2]_.
+
+    The estimate of the variance that we use is computed from the residuals
+    of the least-squares regression and not from the innovations algorithm.
+    This is because our fast implementation of the innovations algorithm is
+    only valid for stationary processes, and the Hannan-Rissanen procedure may
+    produce estimates that imply non-stationary processes. To avoid
+    inconsistency, we never compute this latter variance here, even if it is
+    possible. See test_hannan_rissanen::test_brockwell_davis_example_517 for
+    an example of how to compute this variance manually.
+
+    This procedure assumes that the series is stationary.
+
+    TODO: better warnings / errors when series are short relative to lag length
+
+    TODO: add support for fixed parameters
+
+    TODO: seems like we could add support for seasonal parameters here?
```

Comment:
same question about placement of TODO comments.  (going to omit this if I see it below)

Reply:
Thanks.